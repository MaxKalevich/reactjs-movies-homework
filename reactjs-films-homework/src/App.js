import MainPage from "./pages/main-page/MainPage";
import MovieDetailsPage from "./pages/movie-details-page/MovieDetailsPage";
import ActorProfilePage from "./pages/actor-profile-page/ActorProfilePage";
import Header from "./components/header/Header";

import { ErrorBoundary } from "./components/error-fallback/ErrorFallback";

import "./app-style.scss";

const App = () => (
  <>
    <ErrorBoundary>
      <Header />
      <div className="app-wrapper">
        <MainPage />
        {/*<MovieDetailsPage />*/}
        {/*<ActorProfilePage />*/}
      </div>
    </ErrorBoundary>
  </>
);

export default App;
