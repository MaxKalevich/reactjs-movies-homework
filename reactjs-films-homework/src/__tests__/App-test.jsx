import { render, screen } from "@testing-library/react";
import App from "../App";

describe("Checking App component", () => {
  it("renders App component", () => {
    render(<App />);
    // screen.debug();
  });
  it("snapshot App component", () => {
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot();
  });
});
