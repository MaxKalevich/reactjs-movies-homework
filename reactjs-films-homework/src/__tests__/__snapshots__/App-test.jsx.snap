// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking App component snapshot App component 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <div
      class="header-wrapper"
    >
      <h1
        class="heading"
      >
        Title
      </h1>
      <div
        class="header-group"
      >
        <div
          class="search"
        >
          <input
            class="search-bar"
            id="search"
            placeholder="Movies, person, movie theaters"
            type="text"
            value=""
          />
          <img
            alt="search icon"
            class="search-icon"
            src="search-zoom.svg"
          />
        </div>
        <select
          class="lang-btn"
          id="select-language"
          size="1"
        >
          <option
            selected=""
            value="en"
          >
            En
          </option>
          <option
            value="ru"
          >
            Ru
          </option>
        </select>
      </div>
    </div>
  </header>
  <div
    class="app-wrapper"
  >
    <div
      class="movie-details-page__content-container"
    >
      <div
        class="movie-banner-info-container"
      >
        <div
          class="movie-banner"
        >
          <span
            class="estimation-movie-banner"
          >
            4.2
          </span>
          <img
            alt="movie-banner"
            class="movie-banner__movie-img"
            src="movie-picture-mock.jpg"
          />
        </div>
        <div
          class="movie-info__movie-information"
        >
          <h1
            class="movie-info__movie-title"
          >
            Spider Man
          </h1>
          <h3>
            Description
          </h3>
          <p
            class="movie-info__movie-description"
          >
            Отвязный и вместе с тем душевный кинокомикс Джеймса Ганна теперь в подпискезный и вместе с тем душевный кинокомикс Джеймса Ганзный и вместе с тем душевный кинокомикс Джеймса Ганзный и вместе с тем душевный кинокомикс Джеймса Ганзный и вместе с тем душевный кинокомикс Джеймса Ган.
          </p>
          <h3>
            Release Date
          </h3>
          <p
            class="movie-info__release-date"
          >
            10.10.2022
          </p>
          <h3>
            Revenue
          </h3>
          <p
            class="movie-info__release-revenue"
          >
            $10000000000
          </p>
          <h3>
            Duration
          </h3>
          <p
            class="movie-info__release-duration"
          >
            2:12
          </p>
        </div>
      </div>
      <div
        class="actor-images-group"
      >
        <h2
          class="actor-top-cast-header"
        >
          Top Billed Cast
        </h2>
        <div
          class="actor-card-container"
        >
          <div
            class="actor-card"
          >
            <img
              alt="actor"
              class="actor-tab"
              src="./actor-picture-mock.jpg"
            />
            <div
              class="actor-inf"
            >
              <h2
                class="actor-name"
              >
                Daniel Radcliff
              </h2>
              <p
                class="actor-role"
              >
                Main person
              </p>
            </div>
          </div>
          <div
            class="actor-card"
          >
            <img
              alt="actor"
              class="actor-tab"
              src="./actor-picture-mock.jpg"
            />
            <div
              class="actor-inf"
            >
              <h2
                class="actor-name"
              >
                Daniel Radcliff
              </h2>
              <p
                class="actor-role"
              >
                Main person
              </p>
            </div>
          </div>
          <div
            class="actor-card"
          >
            <img
              alt="actor"
              class="actor-tab"
              src="./actor-picture-mock.jpg"
            />
            <div
              class="actor-inf"
            >
              <h2
                class="actor-name"
              >
                Daniel Radcliff
              </h2>
              <p
                class="actor-role"
              >
                Main person
              </p>
            </div>
          </div>
          <div
            class="actor-card"
          >
            <img
              alt="actor"
              class="actor-tab"
              src="./actor-picture-mock.jpg"
            />
            <div
              class="actor-inf"
            >
              <h2
                class="actor-name"
              >
                Daniel Radcliff
              </h2>
              <p
                class="actor-role"
              >
                Main person
              </p>
            </div>
          </div>
          <div
            class="actor-card"
          >
            <img
              alt="actor"
              class="actor-tab"
              src="./actor-picture-mock.jpg"
            />
            <div
              class="actor-inf"
            >
              <h2
                class="actor-name"
              >
                Daniel Radcliff
              </h2>
              <p
                class="actor-role"
              >
                Main person
              </p>
            </div>
          </div>
          <div
            class="actor-card"
          >
            <img
              alt="actor"
              class="actor-tab"
              src="./actor-picture-mock.jpg"
            />
            <div
              class="actor-inf"
            >
              <h2
                class="actor-name"
              >
                Daniel Radcliff
              </h2>
              <p
                class="actor-role"
              >
                Main person
              </p>
            </div>
          </div>
        </div>
        <h2
          class="movie-details-image-header"
        >
          Images
        </h2>
        <div
          class="images"
        >
          <div
            class="image-container"
          >
            <img
              alt=""
              class="image-banner"
              src="movie_image-picture-mock.jpg"
            />
          </div>
          <div
            class="image-container"
          >
            <img
              alt=""
              class="image-banner"
              src="movie_image-picture-mock.jpg"
            />
          </div>
          <div
            class="image-container"
          >
            <img
              alt=""
              class="image-banner"
              src="movie_image-picture-mock.jpg"
            />
          </div>
          <div
            class="image-container"
          >
            <img
              alt=""
              class="image-banner"
              src="movie_image-picture-mock.jpg"
            />
          </div>
          <div
            class="image-container"
          >
            <img
              alt=""
              class="image-banner"
              src="movie_image-picture-mock.jpg"
            />
          </div>
          <div
            class="image-container"
          >
            <img
              alt=""
              class="image-banner"
              src="movie_image-picture-mock.jpg"
            />
          </div>
          <div
            class="image-container"
          >
            <img
              alt=""
              class="image-banner"
              src="movie_image-picture-mock.jpg"
            />
          </div>
        </div>
      </div>
      <h2
        class="recommendations-header"
      >
        Recommendations
      </h2>
      <div
        class="recommendations"
      >
        <div
          class="card"
        >
          <span
            class="estimation-movie"
          >
            2.4
          </span>
          <img
            alt=""
            class="movie-image"
            src="movie-picture-mock.jpg"
          />
          <div
            class="movie-info"
          >
            <h2
              class="movie-title-card"
            >
              Transporter
            </h2>
            <p
              class="movie-description-main-page"
            />
            <span
              class="movie-genre"
            >
              Horror
            </span>
          </div>
        </div>
        <div
          class="card"
        >
          <span
            class="estimation-movie"
          />
          <img
            alt=""
            class="movie-image"
            src="movie-picture-mock.jpg"
          />
          <div
            class="movie-info"
          >
            <h2
              class="movie-title-card"
            />
            <p
              class="movie-description-main-page"
            />
            <span
              class="movie-genre"
            />
          </div>
        </div>
        <div
          class="card"
        >
          <span
            class="estimation-movie"
          />
          <img
            alt=""
            class="movie-image"
            src="movie-picture-mock.jpg"
          />
          <div
            class="movie-info"
          >
            <h2
              class="movie-title-card"
            />
            <p
              class="movie-description-main-page"
            />
            <span
              class="movie-genre"
            />
          </div>
        </div>
        <div
          class="card"
        >
          <span
            class="estimation-movie"
          />
          <img
            alt=""
            class="movie-image"
            src="movie-picture-mock.jpg"
          />
          <div
            class="movie-info"
          >
            <h2
              class="movie-title-card"
            />
            <p
              class="movie-description-main-page"
            />
            <span
              class="movie-genre"
            />
          </div>
        </div>
        <div
          class="card"
        >
          <span
            class="estimation-movie"
          />
          <img
            alt=""
            class="movie-image"
            src="movie-picture-mock.jpg"
          />
          <div
            class="movie-info"
          >
            <h2
              class="movie-title-card"
            />
            <p
              class="movie-description-main-page"
            />
            <span
              class="movie-genre"
            />
          </div>
        </div>
        <div
          class="card"
        >
          <span
            class="estimation-movie"
          />
          <img
            alt=""
            class="movie-image"
            src="movie-picture-mock.jpg"
          />
          <div
            class="movie-info"
          >
            <h2
              class="movie-title-card"
            />
            <p
              class="movie-description-main-page"
            />
            <span
              class="movie-genre"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
